### PR TITLE
Corner steering updates

### PR DESCRIPTION
**Pull request by kevincloutier**
@vssystemluba , @ejunkins This is the "Corner Steering Update", it's fairly significant and is going to require some effort for review. This includes changes to the parts list (motor, mount, standoffs, etc.); documentation steps; images used in the documentation; new CAD models; and much more. I have tried to test, retest, and test again but having another set of eyes on this will be much appreciated. If you need me to make changes etc. before or after integration, no worries at all, I'm more interested in getting this right then anything else. - Kevin

### CODE REVIEW

**Comment by ericjunkins** [comment]
@kevincloutier I will try and look over this stuff in the next few days and give any feedback about it if necessary, at first glance looks really nice and well done though good job! :)

**Comment by ericjunkins** [comment]
@kevincloutier I was just looking over the CAD files for this, it looks like you did your CAD editing in AutoDesk correct? There is a .SMT file for the corner steering assembly, which I believe is for AutoDesk. I also need to point out that the Final Rover CAD assembly in the Mechanical Integration folder is not effected by your changes, meaning this would look to have two different flavors of the corner steering module.

Because of these I think we should try and come up with a good solution on managing CAD files in the future, and think about how to most easily accommodate users. Below are some of my concerns we might want to figure out how to address

  * Current CAD is done in SolidWorks, which most do not have access to. We provide .STEP files but this eventually leads to incompatibilities, as we can see already in the Mechanical Integration. Does it make sense to attempt a move to an open-source CAD platform instead?
  * Mechanical Integration final rover assembly is not dynamically linked to any of the sub-assembly folders. I meant each of those folders in the /mechanical/ folder to be self-contained and not have relative links between them, but maybe there is a better structure for this? 
  * Overall Location of CAD files - I wonder if there is a better structure for location of CAD files, such that you can download all the CAD separately from build documentation, similar to how the code files work. My hesitation is having a mismatch of CAD-to-build doc happening, but could also have potential organization benefits worth thinking about. 

These certainly don't have to be solved here in this pull request, but things to think about brought about from looking at this. I think to merge this for CAD let's just make sure there is an accurate STEP file of the entire rover with new corner modules generated, and remove the current rover solidworks assembly?
@vssystemluba  what are your thoughts?

**Comment by vssystemluba** [comment]
FYI, I'm currently looking through the documentation changes now and will update with a couple minor edits, but so far things look great.

As for the CAD changes, for this pull request let's simply focus on making sure that the original file formats that we had in the repo (.sldprt, .sldasm, .step, .igs, and 3d view.pdf) are still supported, and that the relevant CAD in the subassemblies as well as the CAD for the full rover model are all correct in each of the above formats.  Let's create a separate issue to address other options for managing CAD changes in the future.

**Comment by ericjunkins** [comment]
@vssystemluba For the CAD files to be the same someone is going to have to rebuild the CAD changes @kevincloutier made but SolidWorks instead of AutoDesk is what i'm saying though if that's what we want to have happen

**Comment by kevincloutier** [comment]
@vssystemluba @ericjunkins Hey guys, sorry for the late reply, I was traveling most of the day. After reading @ericjunkins comments it's clear to me that there is some more work here. But I agree with @vssystemluba that it should be a separate issue, one which is important and should be tackled in the next few weeks. I'm going to toss some of my thoughts here, but maybe we can discuss on the call as well?

1. **SolidWorks** - Yes, I used AutoDesk's Fusion360 as they have a "maker" license which is free. From a very unscientific survey, I find this is becoming the de facto standard for open source projects like this. SolidWorks is prohibitively expensive for hobbyist, secondary schools, etc. Though most college engineering programs provide access. So, in order to allow members of the community to edit these files, we likely need a more moderately priced solution.

2. **Mechanical Integration** - This is a result of the fact I haven't gotten to that step yet, so I wasn't even thinking "if I change the steering mechanism here, I need to updated mechanical integration as well". How do you guys want to handle that? In this PR or a separate one?

3. **Overall Location of CAD Files** - This is similar to the issue @ericjunkins and I were emailing about last week, subjectively I feel that the source code is a fairly large barrier to entry for the project. How the resources are stored and structured makes it difficult for the uninitiated to get started. This is NOT to say the structure is bad or wrong, but that it is tough for someone without our backgrounds and experience to grasp this project quickly (I'm thinking of the high school robotics club, not engineers like us). I'm not sure changing the structure is the answer, but rather that providing a different point of entry for individuals who aspire to be engineers may help. I don't want to beat this issue to death, but maybe it's one we should explore a bit on the phone with the entire team. There is not perfect solution here, but I think we could be better.

4. **Format of CAD Files** - Agreed...I think this is a "growing pain". When it was just the two of you it was likely fairly easy (well, easier anyway) to communicate amongst one another about standards, etc. but now we may need to be more explicit and think about how we can enforce some standards using our tools, build processes, etc.

To sum it up, this PR brings to light a bunch of "stuff" as @ericjunkins rightly points out, but if we start planning how to tackle each of these I think we can overcome them all, and do so fairly quickly.

**Comment by vssystemluba** [comment]
@ericjunkins @kevincloutier I've reviewed the build documentation and images and made a couple minor edits that I believe improve clarity.  I also fixed a quantity inconsistency with the new standoffs (they are sold in packs of 4) and rebuilt the master parts list.

I'm all finished and approve of merging this into master once @ericjunkins incorporates the updated CAD.

Great work again everyone!!

**Comment by kevincloutier** [comment]
I know it's lame, but Teamwork! I'm psyched how this all came together.

**Comment by ericjunkins** [comment]
Okay I added the CAD files to your branch. I made a mistake when switching between credentials so the commit is not tagged to me. Everything for this should be good now so merging this in with master! Thanks for the work on this everyone
